### PR TITLE
Load extensions from the same directory as nuget.exe

### PR DESCRIPTION
**Pull request by zarenner**
Fixes https://github.com/NuGet/Home/issues/1576

### CODE REVIEW

File: src/NuGet.Clients/NuGet.CommandLine/Program.cs [hunk 1]
```diff
@@ -257,6 +257,13 @@ private static void AddExtensionsToCatalog(AggregateCatalog catalog, IConsole co
                     RegisterExtensions(catalog, files, console);
                 }
             }
+
+            // Ideally we want to look for all files. However, using MEF to identify imports results in assemblies being loaded and locked by our App Domain
+            // which could be slow, might affect people's build systems and among other things breaks our build.
+            // Consequently, we'll use a convention - only binaries ending in the name Extensions would be loaded.
```
Comment (yishaigalatzer) [inline src/NuGet.Clients/NuGet.CommandLine/Program.cs:263]: this needs a doc.nuget.org change

**Comment by yishaigalatzer** [comment]
:shipit: from me, @deepakaravindr please verify/review and merge

**Comment by zarenner** [comment]
Hi @deepakaravindr, any idea when this will be merged in?  Just want to know for tracking on our side.

**Comment by zarenner** [comment]
Ping. @deepakaravindr @yishaigalatzer.

**Comment by yishaigalatzer** [comment]
@zarenner nice, I was just walking over to @deepakaravindr to ask the same q.

**Comment by deepakaravindr** [comment]
I have assigned this PR to myself. Looking at it now..

**Comment by deepakaravindr** [comment]
Verified locally that the changes are good and works as desired

image

**Comment by deepakaravindr** [comment]
:shipit: (I will merge it once I get another sign-off)

**Comment by emgarten** [comment]
:shipit:

**Comment by deepakaravindr** [comment]
This has been merged: https://github.com/NuGet/NuGet.Client/commit/4c645bb6b3f3ce685407dbb8e2790cdc4ab3f576

**Pull request closed without merging**